### PR TITLE
Firmware uploading

### PR DESCRIPTION
**Pull request by tannermpeterson**

### CODE REVIEW

File: deployments/apiv2/services/mantarray/src/main.py [hunk 1]
```diff
@@ -124,55 +142,220 @@ async def delete_serial_number(
...
     except Exception:
         err_msg = f"{fw_type.title()} Firmware v{version} not found"
         logger.exception(err_msg)
         return JSONResponse(status_code=status.HTTP_404_NOT_FOUND, content={"message": err_msg})
+
+
+@app.post("/firmware/{fw_type}/{version}")
+async def upload_firmware_file(
+    request: Request,
+    details: MainFirmwareUploadRequest | ChannelFirmwareUploadRequest,
```
Comment (luciipak) [inline deployments/apiv2/services/mantarray/src/main.py:253]: What's the difference again between main firmware and channel firmware?

File: frontend/pulse3d/src/pages/production-console.js [hunk 2]
```diff
@@ -1,22 +1,725 @@
...
+  return (
+    <TableContainer>
+      <TableHeader>Serial Numbers</TableHeader>
       <SerialNumberTable />
-    </Container>
+    </TableContainer>
+  );
+}
+
+export default function ProductionConsole() {
```
Comment (luciipak) [inline frontend/pulse3d/src/pages/production-console.js:708]: Do these all need to go on the same page or can we split them out so it's a bit easier to follow?

File: frontend/pulse3d/src/pages/production-console.js [hunk 3]
```diff
@@ -1,22 +1,725 @@
...
+  };
+
+  const getFileName = (file) => {
+    return file ? file.name : null;
+  };
+
+  const noErrors = () => {
+    return Object.values(uploadOptionErrors).every((val) => val.length === 0);
+  };
+  fwTypes.includes(uploadOptions.fwType);
```
Comment (luciipak) [inline frontend/pulse3d/src/pages/production-console.js:211]: is this meant to be here?

File: deployments/apiv2/services/mantarray/src/main.py (hunk 1 above)
Comment (tannermpeterson) [reply deployments/apiv2/services/mantarray/src/main.py:253]: The main FW runs only on the main microcontroller of the instrument, the channel FW runs on each of the microcontrollers under each well

File: frontend/pulse3d/src/pages/production-console.js (hunk 2 above)
Comment (tannermpeterson) [reply frontend/pulse3d/src/pages/production-console.js:708]: These are probably fine all being on the same page

File: frontend/pulse3d/src/pages/production-console.js (hunk 3 above)
Comment (tannermpeterson) [reply frontend/pulse3d/src/pages/production-console.js:211]: Not sure, looks like it's not doing anything though

File: frontend/pulse3d/src/pages/production-console.js (hunk 2 above)
Comment (luciipak) [reply frontend/pulse3d/src/pages/production-console.js:707]: I mean all the components, can we break them up so it's not such a large file?

Comment (tannermpeterson) [reply frontend/pulse3d/src/pages/production-console.js:707]: Yeah I can split it up

Comment (tannermpeterson) [reply frontend/pulse3d/src/pages/production-console.js:707]: Actually I think that might cause problems with the styling. I can make a note to break everything up once we find a way to reuse all the styling

**Review by luciipak — APPROVED** [review]

**Pull request merged**